File: src/main/java/com/amm/kafkafirst/interfaces/ConsumerController.java
package com.amm.kafkafirst.interfaces;

import com.amm.kafkafirst.application.ConsumerService;
import com.amm.kafkafirst.application.ProducerService;
import com.amm.kafkafirst.infrastructure.Message;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;

@Controller
public class ConsumerController {

    private static final Logger logger = LoggerFactory.getLogger(ConsumerController.class);

    private ConsumerService consumerService;

    public ConsumerController(@Autowired ConsumerService consumerService){
        this.consumerService = consumerService;
    }

    @GetMapping("/consumer")
    public ResponseEntity<HttpStatus> index() {
        // Message message = new Message(200, "Hello World");
        logger.info(">>> {} ", consumerService.getMessages().toString());
        return new ResponseEntity<>(HttpStatus.OK);
    }
}

File: src/main/kotlin/com/amm/kafkafirst/infrastructure/interfaces/ConsumerController.kt
package com.amm.kafkafirst.infrastructure.interfaces;

import com.amm.kafkafirst.infrastructure.services.ConsumerService;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class ConsumerController {

    private static final Logger logger = LoggerFactory.getLogger(ConsumerController.class);

    private ConsumerService consumerService;

    public ConsumerController(@Autowired ConsumerService consumerService){
        this.consumerService = consumerService;
    }

    @GetMapping("/consumer")
    public ResponseEntity<HttpStatus> index() {
        logger.info(">>> {} ", consumerService.getMessages().toString());
        return new ResponseEntity<>(HttpStatus.OK);
    }
}